### PR TITLE
Fix being unable to set active creds, if there has never been any

### PR DESCRIPTION
**Pull request by abrooksv**
Also cache the lookup of credential factories

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
If the active cred provider was null (never set, no default profile), we would crash when trying to determine which of the credentials should be selected in the pop ups.

If there was no default profile, we also paid an expensive cost (and locked up the UI)  due to re-creating all the credential providers and parsing of the profile file for each possible profile. We instead cache the profile factories so they are only ever created once.

## Related Issue(s)
#563

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsSettingsPanel.kt [hunk 1]
```diff
@@ -214,7 +214,11 @@ private class ChangeRegionAction(val region: AwsRegion) : ToggleAction(region.di
 
 private class ChangeCredentialsAction(val credentialsProvider: ToolkitCredentialsProvider) : ToggleAction(credentialsProvider.displayName), DumbAware {
 
-    override fun isSelected(e: AnActionEvent): Boolean = getAccountSetting(e).activeCredentialProvider == credentialsProvider
+    override fun isSelected(e: AnActionEvent): Boolean = try {
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsSettingsPanel.kt]: another way to do this is 

```
tryOrNull { getAccountSetting(e).activeCredentialProvider == credentialsProvider } ?: false
```

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ExtensionPointCredentialsProviderRegistry.kt [hunk 2]
```diff
@@ -33,20 +33,24 @@ class CredentialProviderFactoryEP : AbstractExtensionPointBean() {
...
     companion object {
         private val LOG = Logger.getInstance(ExtensionPointCredentialsProviderRegistry::class.java)
         private const val EP_NAME = "aws.toolkit.credentialProviderFactory"
-        val EXTENSION_POINT = ExtensionPointName.create<CredentialProviderFactoryEP>(EP_NAME)
+        private val EXTENSION_POINT = ExtensionPointName.create<CredentialProviderFactoryEP>(EP_NAME)
+
+        private val factories by lazy {
+            EXTENSION_POINT.extensions
+                .mapNotNull {
+                    try {
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ExtensionPointCredentialsProviderRegistry.kt]: `LOG.tryOrNull { it.getHandler() }`

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ExtensionPointCredentialsProviderRegistry.kt]: That is an extension on slf4j, not the IJ diagnostic Logger

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ExtensionPointCredentialsProviderRegistry.kt]: Guess using slf4j isnt a big deal here

Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ExtensionPointCredentialsProviderRegistry.kt]: one just delegates to the other so it's much of a muchness.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsSettingsPanel.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsSettingsPanel.kt]: Done

**Comment by mogren** [comment]
Thanks for the quick fix!

**Pull request merged**